src/plugins/data/common/utils/helpers.ts

```diff
@@ -42,29 +40,27 @@ export interface QueryStatusOptions {
   interval?: number;
 }
 
-export const handleQueryResults = <T>(
+export const delay = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+export const handleQueryResults = async <T>(
   options: QueryStatusOptions
 ): Promise<FetchStatusResponse> => {
   const { pollQueryResults, interval = 5000, queryId } = options;
+  let queryResultsRes: FetchStatusResponse;
+  let queryStatus;
+  do {
+    // Wait for the given interval in ms before polling for the query status/results
+    await delay(interval);
+    queryResultsRes = await pollQueryResults();
+    queryStatus = queryResultsRes?.status?.toUpperCase();
+  } while (queryStatus !== 'SUCCESS' && queryStatus !== 'FAILED');
+
+  if (queryStatus === 'FAILED') {
+    throw (
+      (queryResultsRes as QueryFailedStatusResponse).body.error ??
+      new Error(`Failed to fetch results ${queryId ?? ''}`)
```

Comment:
Is this somehow interfaced to the user? If so, we need to use i18n.

Reply:
addressed